src/vm/eventpipe.cpp

```diff
@@ -732,66 +533,68 @@ void EventPipe::WriteEventInternal(EventPipeEvent &event, EventPipeEventPayload
         NOTHROW;
         GC_NOTRIGGER;
         MODE_ANY;
+        PRECONDITION(s_pConfig != nullptr);
+        PRECONDITION(s_pSessions != nullptr);
     }
     CONTRACTL_END;
 
+    if (s_pConfig == nullptr || s_pSessions == nullptr)
```

Comment:
@cshung - Not a high pri issue, but one spot that might be interesting to poke for reliability is synchronization between writer threads and shutdown. Any potential issues here pre-dated @jorive 's change but the fact that Brian originally put similar checks in the code suggests that some thread could try to call WriteEvent() after EventPipe::Shutdown() had occured. And if the call can come afterwards, there is a decent chance it could also happen concurrently. Even if s_pConfig isn't nullptr at this instruction, maybe it becomes null 20 lines lower in the function and I'm guessing the results wouldn't be good : )